app/routes/goal_routes.py

```diff
@@ -1 +1,107 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from datetime import datetime
+from app.routes.utilities_routes import *
+
+bp = Blueprint("goal_bp",__name__, url_prefix= "/goals")
+
+
+####################################################################
+######################### Create FUNCTIONS #########################
+####################################################################
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    
+    if not request_body or not request_body.get("title"):
+        return make_response({"details": "Invalid data: missing title"}, 400) 
+    
+    goal = create_model(Goal, request_body)
+    return make_response(goal, 201)
+
+
+@bp.post("/<goal_id>/tasks")
+def post_task_ids_to_goal(goal_id):
+    request_body = request.get_json() 
+    goal = validate_model(Goal, goal_id)
+
+    task_ids = request_body.get("task_ids", [])
+    for task_id in task_ids:
+        task = validate_model(Task, task_id)
+        goal.tasks.append(task)
+    
+    db.session.commit()
+
+    response_body = { 
+        "id": goal.id, 
+        "task_ids": [task.id for task in goal.tasks]
+    }
+
+
+    return response_body, 200
+
+
+####################################################################
+######################### READ FUNCTIONS #########################
+####################################################################
+
+@bp.get("")
+def get_goals():
+    request_arguements = request.args
+    goals = get_models_with_filters(Goal, request_arguements)
+    if not goals: 
+        return make_response([], 200)
```

Comment:
This check is redundant. `goals` will alreayd be an empty list is nothing was found, so the return on line 58 willwork identically.